specs/dynamicdependencies/sample-2.md

```diff
@@ -0,0 +1,85 @@
+# Sample 2 - Fabrikam app using Contoso's Muffins package with smart class helpers
+
+Variant of Sample 1 using RAII classes to manage packageDependencyId, packageDependencyContext and DLL lifetimes.
+
+```MddPinPackageDependency``` will fail if there are no packages registered for the calling user satisfying the PackageDependency.
+
+## Win32
+
+```c++
+#include <MsixDynamicDependency.hpp>
+#include <wil/resource.h>
+
+class MddPackageDependencyContextAndDll
+{
+public:
+    MddPackageDependencyContextAndDll() = default;
+
+    ~MddPackageDependencyContextAndDll()
+    {
+        (void) LOG_IF_FAILED(MddRemovePackageDependency(m_packageDependencyContext));
+        (void) LOG_IF_WIN32_BOOL_FALSE(FreeLibrary(m_module.get()));
```

Comment:
Why this vs letting `m_module` self-destruct? If you do keep this, use `m_module.release()` so it doesn't double-free it.